app/models/robots/pooling_and_splitting_robot.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+module Robots
+  class PoolingAndSplittingRobot < Robot # rubocop:todo Style/Documentation
+    attr_writer :relationships
+
+    def verified
+      @verified ||= {}
+    end
+
+    #
+    # Returns a hash of bed barcodes and their valid state
+    # Also adds any errors describing invalid bed states
+    #
+    # @return [Hash<String => Boolean>] Hash of boolean indexed by bed barcode
+    def valid_relationships
+      raise StandardError, "Relationships for #{name} are empty" if @relationships.empty?
+
+      @relationships.each do |relationship|
+        parent_beds = relationship.dig('options', 'parents')
+        child_beds = relationship.dig('options', 'children')
+
+        used_parents = []
+        verify_child_beds(child_beds, parent_beds, used_parents)
```

Comment:
Rather than passing in an array to mutate, return the unused parents instead. Mutating arguments can get very confusing.

Reply:
To avoid passing used_parents, child_beds and parent_beds around all over the place I put them up at the top as instance variables.
And then setting the initial value to something with @myvar = value.
Not sure that's best practice though.

Reply:
Lol this happens alot every time some one has @Myvar in there comments i get added to some random thread on github, funny...